File: Labs/Lab1/Lab1.py
# Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to
# target.
# You may assume that each input would have exactly one solution, and you may not use the same element twice.
# You can return the answer in any order.

nums = [1, 3, 5, 7, 9]

target = 6

def term_indices (numbers, sum):
    for i in numbers:
        if sum - i in numbers:
            if numbers.index(i) == numbers.index(sum - 1):
                print("Term indices cannot be repeated")
                return
            else:
                print("Indices: " + str(numbers.index(i)) + ", " + str(numbers.index(sum - i)))
                return
        else:
            print("Not enough terms that can add up to", target)
            return

term_indices(nums, target)

File: Labs/Lab1/Lab1.java
import java.util.*;

public class Lab1 {
    public static void main (String[] args) {
        int[] nums = {1, 3, 5, 7, 9};
        int target = 6;

        termIndices(nums, target);
    }
    public static void termIndices (int[] numbers, int sum) {
        for (int i = 0; i <= numbers.length - 1; i++) {
            if (checkContains(sum -numbers[i], numbers)) {
                if (checkIndex(numbers, numbers[i]) == checkIndex(numbers, sum - numbers[i])) {
                    System.out.println("Term indices cannot be repeated");
                    return;
                }
                else {
                    System.out.println("Indices: ");
                    System.out.println(i);
                    System.out.println(checkIndex(numbers, sum - numbers[i]));
                    return;
                }
            }
            else {
                System.out.println("Not enough terms that can add up to the target");
                return;
            }
        }
    }

    public static boolean checkContains(int term, int[] numbers) {
        boolean contains = false;
        for (int j : numbers) {
            if (j == term) {
                contains = true;
            }
        }
        return contains;
    }

    public static int checkIndex(int[] numbers, int term) {
        if (numbers == null) {
            return -1;
        }

        int length = numbers.length;
        int i = 0;

        while (i < length) {
            if (numbers[i] == term) {
                return i;
            }
            else {
                i += 1;
            }
        }
        return -1;
    }
}
